code/modules/mapping/random_rooms.dm

```diff
@@ -1479,3 +1479,139 @@
 	template_width = 5
+
+/datum/map_template/random_room/skwh_rdm158
+	name = "Large Abandoned Virology"
+	room_id = "skwh_rdm158_abandonedviro"
+	mappath = "_maps/RandomRooms/10x10/skwh_rdm158_abandonedviro.dmm"
+	centerspawner = FALSE
+	template_height = 10
+	template_width = 10
+	weight = 2
+	stock = 3
+
+/datum/map_template/random_room/skwh_rdm157
+	name = "Slime Storage"
+	room_id = "skwh_rdm157_slimestorage"
+	mappath = "_maps/RandomRooms/10x5/skwh_rdm157_slimestorage.dmm"
+	centerspawner = FALSE
+	template_height = 5
+	template_width = 10
+	weight = 3
+
+/datum/map_template/random_room/skwh_rdm159
+	name = "Monkey Testing Room"
+	room_id = "skwh_rdm159_MONKETESTSUBJECT"
+	mappath = "_maps/RandomRooms/10x10/skwh_rdm159_MONKETESTSUBJECT.dmm"
+	centerspawner = FALSE
+	template_height = 10
+	template_width = 10
+	weight = 1
+	stock = 5
+
+/datum/map_template/random_room/skwh_rdm139
+	name = "SAW GBS Maint Game Room"
+	room_id = "skwh_rdm139_GBSsawgameroom"
+	mappath = "_maps/RandomRooms/5x3/skwh_rdm139_GBSsawgameroom.dmm"
+	centerspawner = FALSE
+	template_height = 3
+	template_width = 5
+	weight = 2
+	stock = 1
+
+/datum/map_template/random_room/skwh_rdm160
+	name = "Green Viney Riches"
+	room_id = "skwh_rdm160_greenvineyriches"
+	mappath = "_maps/RandomRooms/10x10/skwh_rdm160_greenvineyriches.dmm"
+	centerspawner = FALSE
+	template_height = 10
+	template_width = 10
+	weight = 6
+
+/datum/map_template/random_room/maskroom3x5
+	name = "Haunted Old Maskroom"
+	room_id = "maskroom3x5"
+	mappath = "_maps/RandomRooms/3x5/maskroom3x5.dmm"
+	centerspawner = FALSE
+	template_height = 5
+	template_width = 3
+	weight = 5
+
+/datum/map_template/random_room/secretivemaintwater
+	name = "Secretive Maint Water Collector Room 1"
+	room_id = "secretivemaintwatercollectorroom"
+	mappath = "_maps/RandomRooms/3x3/secretivemaintwatercollectorroom.dmm"
+	centerspawner = FALSE
+	template_height = 3
+	template_width = 3
+	weight = 4
+	stock = 6
+
+/datum/map_template/random_room/DRUGDEALERHOBOS
+	name = "Hobo Drug Production Room"
+	room_id = "DRUGDEALERHOBOS"
+	mappath = "_maps/RandomRooms/10x5/DRUGDEALERHOBOS.dmm"
+	centerspawner = FALSE
+	template_height = 5
+	template_width = 10
+	weight = 3
+	stock = 5
+
+/datum/map_template/random_room/randomlostartifact
+	name = "Room of the Lost Maint Artifact of OLD"
+	room_id = "randomlostartifact"
+	mappath = "_maps/RandomRooms/5x3/randomlostartifact.dmm"
+	centerspawner = FALSE
+	template_height = 3
+	template_width = 5
+	weight = 3
+	stock = 2
+
+/datum/map_template/random_room/ARTROOM
+	name = "Hazardous Abandoned Artifact Room"
+	room_id = "ARTIFACTROOM"
+	mappath = "_maps/RandomRooms/10x5/ARTIFACTROOM.dmm"
+	centerspawner = FALSE
+	template_height = 5
+	template_width = 10
+	weight = 1
+	stock = 7
+
+/datum/map_template/random_room/BlobInfestedXenobio
+	name = "Abandoned Old Blob Infested Xenobio Room"
+	room_id = "XenobioBlobInfested"
+	mappath = "_maps/RandomRooms/10x10/XenobioBlobInfested.dmm"
```

Comment:
No.
Contains ghost roles.
If you had actually tested this room in-game, you can see the blob actually can spread outside of the room instantly.

Reply:
will say i have tested this room many times and had been trying to reinforce the walls to stronger walls to keep the blob node stuff in, also the ghost role was not intended, but got turned into a ghost role cause one of the blob spores keep taking over the dead corpse in the room, anyways i deleted this room from the PR